Jenkinsfile

```diff
@@ -17,52 +17,35 @@
 @Library('github.com/fabric8io/fabric8-pipeline-library@master')
 def utils = new io.fabric8.Utils()
 def flow = new io.fabric8.Fabric8Commands()
-dockerTemplate{
-   clientsNode {
-    ws{
+
+dockerTemplate {
+  clientsNode {
+    ws {
       checkout scm
 
-      if (utils.isCI()){
-        def snapshotImageName = "fabric8/maven-builder:SNAPSHOT-${env.BRANCH_NAME}-${env.BUILD_NUMBER}"
-        container('docker'){
-          stage('build snapshot image'){
-            sh "docker build -t ${snapshotImageName} ."
-          }
-          stage('push snapshot image'){
-            sh "docker push ${snapshotImageName}"
-          }
-        }
-        stage('notify'){
-            def pr = env.CHANGE_ID
-            if (!pr){
-                error "no pull request number found so cannot comment on PR"
-            }
-            def message = "snapshot maven-builder image is available for testing.  `docker pull ${snapshotImageName}`"
-            container('docker'){
-                flow.addCommentToPullRequest(message, pr, 'fabric8io-images/maven-builder')
-            }
+      def version = utils.isCI() ? "SNAPSHOT-${env.BRANCH_NAME}-${env.BUILD_NUMBER}" : getNewVersion{}
+      def image = "fabric8/maven-builder:${version}"
+
+      echo "Building image ${image}"
+
+      container('docker') {
+
+        stage('Build docker image') {
+          sh "docker build -t ${image} ."
         }
-      } else if (utils.isCD()){
-        
-        def v = getNewVersion{}
-        // stage('tag'){
-        //   container('clients'){
-        //     gitTag{
-        //       releaseVersion = v
-        //       skipVersionPrefix = true
-        //     }
-        //   }
-        // }
 
-        def imageName = "fabric8/maven-builder:${v}"
+        stage('Push docker image') {
+          sh "docker push ${image}"
+        }
+      }
 
-        container('docker'){
-          stage('build image'){
-            sh "docker build -t ${imageName} ."
-          }
-          stage('push image'){
-            sh "docker push ${imageName}"
+      if (utils.isCI()) {
```

Comment:
OFFTOPIC: Just a thought. How about we convert `if (utils.isCI()){ ... }` to 

```groovy
mode('ci') {
  // do ci mode steps
}

mode('cd') {
 // do cd mode steps
}
```

Reply:
@sthaha I've felt the same. It could be something first class like 

```
ci {

}

cd {

}
```
as well; but we need to define what those mean and maybe make it configurable for users. For example, some prefer to use a branch called 'deploy' for releases and CD.

Reply:
I chose `mode()` instead of `ci {}` since it is easier to write `mode('ci', 'cd') {  }` to apply to both modes. 

to address the definition part, I think we should be able to do some
1.  groovy meta programming such users can invoke the using `with`
2. or pass in something like 
```groovy

def detectMode(mode) {
  switch mode {
  case 'ci': return Utils.isCI()
  case 'cd' return getCurrentBranch == 'deploy'
  }
}
mode('ci', 'cd',  findMode:  detectMode) {
}

```